### PR TITLE
Support configurable extra plugin lookup path

### PR DESCRIPTION
**Pull request by houqp**
This feature can be handy for platforms that does custom packaging like android, where plugins directory is not accessible from users and wiped every time the package is updated.

### CODE REVIEW

File: frontend/ui/elements/reader_menu_order.lua [hunk 1]
```diff
@@ -23,7 +23,6 @@ local order = {
         "highlight_options",
         "change_font",
         "hyphenation",
-        "read_timer",
```
Comment (Hzj-jie) [inline frontend/ui/elements/reader_menu_order.lua]: Any reason to prefer this plugin to be placed in tools? IMO, it's more like a "reading experience".

Comment (Frenzie) [inline frontend/ui/elements/reader_menu_order.lua]: It does seem a touch out of place with the other rendering options. There should at least be a separator in between if it goes in the typeset menu. I didn't want to make things controversial by changing the menu around (and I didn't have time to think about it anyway) but it should be something more like

```lua
        -- rendering stuff
        "page_overlap",
        "switch_zoom_mode",
        "set_render_style",
        --sep, I don't think highlight options go with anything else really
        "highlight_options",
        --sep, font-related stuff
        "change_font",
        "floating_punctuation",
        "hyphenation",
        --sep, "reading experience" stuff
        "read_timer"
```

Comment (houqp) [inline frontend/ui/elements/reader_menu_order.lua]: Firstly, for consistent reason. It's placed under tools tab in filemanager. We should either keep it consistent between apps or only enable it in reader.

Secondly, that tab is named typeset, which means anything that controls rendering results. A timer doesn't fit into this category. With the new UX proposed by @baskerville and the new separator feature, we can combine navigation and typeset into one read menu, then it makes more sense to put timer there.

File: frontend/pluginloader.lua [hunk 2]
```diff
@@ -1,42 +1,66 @@
...
-            local package_path = package.path
-            local package_cpath = package.cpath
-            package.path = path.."/?.lua;"..package.path
-            package.cpath = path.."/lib/?.so;"..package.cpath
-            local ok, plugin_module = pcall(dofile, mainfile)
-            if not ok or not plugin_module then
-                logger.warn("Error when loading", mainfile, plugin_module)
-            elseif type(plugin_module.disabled) ~= "boolean" or not plugin_module.disabled then
+    local lookup_path_list = { DEFAULT_PLUGIN_PATH }
+    local extra_paths = G_reader_settings:readSetting("extra_plugin_paths")
```
Comment (Hzj-jie) [inline frontend/pluginloader.lua]: Should we directly add DataStorage:getDataDir()/plugins/ to this list.

Comment (houqp) [inline frontend/pluginloader.lua]: I actually don't really expect many people to use this feature right now. We can always add a default path to the list if it turns out to be a popular feature. My preference is to not introduce this extra lookup overhead to all users for now.

**Comment by Hzj-jie** [comment]
Looks good to me.

**Pull request merged**